client_code/Pivot/__init__.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-License-Identifier: MIT
+#
+# Copyright (c) 2021 The Anvil Extras project team members listed at
+# https://github.com/anvilistas/anvil-extras/graphs/contributors
+#
+# This software is published at https://github.com/anvilistas/anvil-extras
+import anvil
+import anvil.js
+
+from ..utils import _component_helpers as helpers
+from ._anvil_designer import PivotTemplate
+
+__version__ = "1.6.0"
+
+cdn = "https://cdnjs.cloudflare.com/ajax/libs"
+pivottable_version = "2.23.0"
+jqueryui_version = "1.11.4"
+tags = [
+    (f"{cdn}/pivottable/{pivottable_version}/pivot.min.css", "link"),
+    (f"{cdn}/jqueryui/{jqueryui_version}/jquery-ui.min.js", "script"),
+    (f"{cdn}/pivottable/{pivottable_version}/pivot.min.js", "script"),
+]
+
+_jquery = anvil.js.window.jQuery
+
+
+try:
+    assert "pivotUtilities" in _jquery.keys()
+except AssertionError:
+    for tag in tags:
+        helpers.add_html_tag(*tag)
+    assert "pivotUtilities" in _jquery.keys()
+
+
+class Pivot(PivotTemplate):
+    option_names = {
+        "rows": "rows",
+        "columns": "cols",
+        "values": "vals",
+        "aggregator": "aggregatorName",
+    }
+
+    def __init__(self, **properties):
+        self.pivot_options = {
+            option: properties[option] for option in self.option_names
+        }
+        spacer = anvil.Spacer()
+        self.node_id = helpers._get_dom_node_id(spacer)
```

Comment:
These methods won't be (easily) available in the Designer.
consider using a div in your HTML with an identifiable class rather than a spacer.
then use

```
dom_node = get_dom_node(self)
self.pivot_node = dom_node.querySelector(".anvil-extras-pivot")
```

This will make it much easier to do a design version

Reply:
and how would that cope with more than one instance on a form?

Reply:
Perhaps use jQuery to add the div with a unique id?

Reply:
we don't need to worry because when we do `dom_node.querySelector` we're only searching in our child elements.

Reply:
ahhhh! of course!